### PR TITLE
To fix Restore can fail when at least one partition was empty #17

### PR DESCRIPTION
**Pull request by rohan26may**

### CODE REVIEW

**Review by LBmaster001 — CHANGES_REQUESTED** [review]
Why Use the Old Java IO, if we already have the NIO used ?
`if (Files.size(topicDir.resolve(segmentFilePrefix) > 0) {
    currentSegment.seek(offset);
}`

File: src/main/java/de/azapps/kafkabackup/common/partition/PartitionReader.java [hunk 1]
```diff
@@ -46,7 +46,9 @@ public void seek(long offset) throws PartitionIndex.IndexException, IOException,
         partitionIndex.seek(offset);
         String segmentFilePrefix = partitionIndex.readFileName();
         currentSegment = new SegmentReader(topic, partition, topicDir, segmentFilePrefix);
-        currentSegment.seek(offset);
+        if (new java.io.File(topicDir.toFile(), segmentFilePrefix).length() != 0) {
```
Comment (itadventurer) [inline src/main/java/de/azapps/kafkabackup/common/partition/PartitionReader.java:49]: I am thinking whether we should check this here or whether the `SegmentReader` is the correct place. Any ideas from your side?

**Comment by itadventurer** [comment]
I like Olofs approach more: https://github.com/getdreams/kafka-backup/pull/6/files Any objections?

**Comment by LBmaster001** [comment]
I don't realy like it, because for me it looks like some kind of workaround and does not fix the real issue here. I tried to explain what I mean here: https://github.com/itadventurer/kafka-backup/issues/17#issuecomment-594429997
Maybe I'm wrong, but this seems to be the better approach for me

Edit: I think I misunderstood how the thing worked at https://github.com/itadventurer/kafka-backup/blob/3c95089c699af6b42207925b05ff04a721da95e1/src/main/java/de/azapps/kafkabackup/common/partition/PartitionWriter.java#L34
The one of Olofs makes much more sense now. LGTM

**Comment by itadventurer** [comment]
Merged Olofs solution ;) Thank you for your help!

**Pull request closed without merging**